tests/MassTransit.SqlTransport.Tests/PgSql/ChannelName_Specs.cs

```diff
@@ -0,0 +1,36 @@
+namespace MassTransit.DbTransport.Tests.PgSql;
+
+using System.Threading.Tasks;
+using NUnit.Framework;
+using SqlTransport.PostgreSql.Helpers;
+
+[TestFixture]
+public class ChannelName_Specs
+{
+    [TestCase("string that has 40 characters 123456789")]
+    public async Task Should_sanitize_schema(string schema)
+    {
+        var sanitizedSchema = NotifyChannel.SanitizeSchemaName(schema);
+
+        Assert.That(sanitizedSchema, Has.Length.EqualTo(39));
+    }
+
+    [TestCase("string that has 38 characters 1234567")]
+    [TestCase("string that has 39 characters 12345678")]
+    public async Task Should_not_sanitize_schema(string schema)
+    {
+        var sanitizedSchema = NotifyChannel.SanitizeSchemaName(schema);
+
+        Assert.That(sanitizedSchema, Is.EqualTo(schema));
+    }
+
+    [TestCase("")]
+    [TestCase(" ")]
+    [TestCase(null)]
```

Comment:
I didn’t test for that because those are allowed characters if using quotes.
The only issue is the string length.

Also if there should be any schema validation, that should be in the options validation in my opinion and not when sanitizing tor the channel name.